### ISSUE TITLE
Framework causes panic during Set Validation

### ISSUE THREAD

**Issue by dglsparsons** [hashicorp/terraform-plugin-framework#261]
### Module version

```
github.com/hashicorp/terraform-plugin-framework v0.5.0
```

### Relevant provider source code

```go
func (r resourceProjectType) GetSchema(_ context.Context) (tfsdk.Schema, diag.Diagnostics) {
	return tfsdk.Schema{
		Attributes: map[string]tfsdk.Attribute{
			// ... - extra attributes omitted
			"name": {
				Required: true,
				Type:     types.StringType,
				Validators: []tfsdk.AttributeValidator{
					stringLengthBetween(1, 52),
				},
				Description: "The desired name for the project.",
			},
			"environment": {
				Description: "A list of environment variables that should be configured for the project.",
				Optional:    true,
				Attributes: tfsdk.ListNestedAttributes(map[string]tfsdk.Attribute{
					"target": {
						Description: "The environments that the environment variable should be present on. Valid targets are either `production`, `preview`, or `development`.",
						Type: types.SetType{ // changing to types.ListType works successfully. 
							ElemType: types.StringType,
						},
						// Validators: []tfsdk.AttributeValidator{
						// 	stringSetItemsIn("production", "preview", "development"),
						// },
						Required: true,
					},
					"key": {
						Description: "The name of the environment variable.",
						Type:        types.StringType,
						Required:    true,
					},
					"value": {
						Description: "The value of the environment variable.",
						Type:        types.StringType,
						Required:    true,
						Sensitive:   true,
					},
					"id": {
						Description: "The ID of the environment variable",
						Type:        types.StringType,
						Computed:    true,
					},
				}, tfsdk.ListNestedAttributesOptions{}),
			},
			"id": {
				Computed: true,
				Type:     types.StringType,
			},
		},
	}, nil
}

```

### Terraform Configuration Files

```hcl
terraform {
  required_providers {
    vercel = {
      source  = "vercel/vercel"
      version = "~> 0.1"
    }
  }
}

// first apply 
resource "vercel_project" "foo" {
  name = "foo"
}

// Remote state shows: 
// $ terraform state show vercel_project.foo
// # vercel_project.foo:
// resource "vercel_project" "foo" {
//     id   = "prj_CSyktdtj1Kw4b4GKgjb4Tabjm2Nv"
//     name = "foo"
// }


// second apply 
resource "vercel_project" "foo" {
  name = "foo"
  environment = [
    {
      key    = "foo"
      value  = "bar"
      target = ["preview", "production"]
    }
  ]
}
```

### Debug Output

https://gist.github.com/dglsparsons/8c5d9da0bc4819f97d5a459450ae7582

### Expected Behavior

The plan/apply output should show the addition of the `environment` array. 
Note: this works if it is the first apply, but does not work if it is applied after 
a previous iteration with no `environment` field set. 

### Actual Behavior

The provider panics with a segfault on `github.com/hashicorp/terraform-plugin-framework/types.SetType.Validate`

### Steps to Reproduce

1. Sign up for a [Vercel](https://vercel.com) account.
2. Generate an API token [from here](https://vercel.com/account/tokens) and set it in an environment variable of `VERCEL_API_TOKEN`. 
1. Put the following config into a terraform file: 
```hcl
terraform {
  required_providers {
    vercel = {
      source  = "vercel/vercel"
      version = "~> 0.1"
    }
  }
}

resource "vercel_project" "foo" {
  name = "foo"
}
```
2. `terraform init`
3. `terraform apply -auto-approve` 
4. Update the config to the following 
```hcl 
terraform {
  required_providers {
    vercel = {
      source  = "vercel/vercel"
      version = "~> 0.1"
    }
  }
}

resource "vercel_project" "foo" {
  name = "foo"
  environment = [
    {
      key    = "foo"
      value  = "bar"
      target = ["preview", "production"]
    }
  ]
}
```
5. `terraform plan` - causes a panic. 

### References

The terraform-provider-vercel source code is available [here](https://github.com/vercel/terraform-provider-vercel)

**Comment by benamib** [comment]
I have the same issue here. I can create my resource with a list but when I try to make an update on that list, it panics.
I tried rolling back to `v0.4.2`, it still fails to update the resource but this time I have this error instead:

```
│ Error: State Read Error
│ 
│   with qovery_application.my_application,
│   on main.tf line 130, in resource "qovery_application" "my_application":
│  130: resource "qovery_application" "my_application" {
│ 
│ An unexpected error was encountered trying to read an attribute from the state. This is always an error in the provider. Please report the following to the provider developer:
│ 
│ ElementKeyInt(1).AttributeName("size") still remains in the path: step cannot be applied to this value

```

**Comment by dglsparsons** [comment]
Okay, so looking into this a bit further, I think the issue is because `Validate()` is called when reading the existing state: 

https://github.com/hashicorp/terraform-plugin-framework/blob/main/tfsdk/state.go#L95

I can see it's trying to validate an `invalid typeless tftypes.Value<> <nil>"` value here.

There's a nicely documened TODO above this call site: 
```go
// TODO: If ErrInvalidStep, check parent paths for unknown value.
//       If found, convert this value to an unknown value.
```

I think this is the issue I'm seeing - it makes sense, as in my example `environment` is initially undefined. Validating `target` when it is present doesn't understand that the parent is unknown. This means it's related to #186. 

I'm unsure of how to resolve this though... :(

**Comment by bflad** [comment]
This should be resolved in https://github.com/hashicorp/terraform-plugin-framework/commit/c584343531cb3bf1b46dad2e87631c75812593c6, which will be released in v0.7.0 of the framework.
